### DISCUSSION TITLE
Delete text (psiElement.delete()) from another plugin and add to vim paste register

### DISCUSSION THREAD

**Discussion by adambasis** [JetBrains/ideavim#881] [Q&A]
Hey, love this plugin :) 👍 

I delete text of a file in my groovy `live-plugin` (https://github.com/dkandalov/live-plugin) with `psiElement.delete()`.
How could I add the deleted text to the default vim `"` register?

Thanks!

**Comment by AlexPl292** [comment]
Hi there. Sorry, we don't yet have a good API layout for IdeaVim. But it sounds like a good case for it.

**Comment by adambasis** [comment]
Yeah no worries. I think the API is fine actually if you use the source code in the repo as a reference.

I accessed ideavim in my live-plugin with `// depends-on-plugin IdeaVIM`, and used the function `putExtensionHandlerMapping` to implement mappings with custom functionality.

I do not use classes. I just call the putMapping functions. So for MappingOwner I used `MappingOwner.IdeaVim.System`, is that fine to use?
